scheduler/src/cook/compute_cluster.clj

```diff
@@ -0,0 +1,105 @@
+         framework-id]}
+  (let [query-result
+        (d/q '[:find [?c]
+             :in $ ?cluster-name? ?mesos-id?
+             :where
+             [?c :compute-cluster/type :compute-cluster.type/mesos]
+             [?c :compute-cluster/cluster-name ?cluster-name?]
+             [?c :compute-cluster/mesos-framework-id ?framework-id?]]
+           unfiltered-db compute-cluster-name framework-id)
+        materialized-query-result (into [] query-result)]
+    materialized-query-result))
+
+(defn get-mesos-cluster-map
+  "Process one mesos cluster specification, returning the entity id of the corresponding compute-cluster, creating the cluster if it does not exist."
+  [conn {:keys [compute-cluster-name framework-id] :as mesos-cluster}]
+  {:pre [compute-cluster-name
+         framework-id]}
+  (let [cluster-entity (get-mesos-cluster-entity (d/db conn) mesos-cluster)]
+    (when (zero? (count cluster-entity))
+      (create-missing-compute-cluster conn (mesos-cluster->compute-cluster-map-for-datomic mesos-cluster)))
+    {:cluster-type :mesos-cluster
+     :compute-cluster-name compute-cluster-name
+     :mesos-framework-id framework-id
+     :db-id (->> (get-mesos-cluster-entity (d/db conn) mesos-cluster)
+                 first)}))
+
+(defn get-mesos-clusters-from-config
+  "Get all of the mesos clusters defined in the configuration.
+  This is a wart that undo'ed the shattering of the :mesos {...} into separate keys that
+  occurs in config.edn, and consolidates it into one dictionary for downstream consumption."
+  [{:keys [mesos-compute-cluster-name mesos-framework-id]}]
+  [{:compute-cluster-name mesos-compute-cluster-name :framework-id mesos-framework-id}])
+
+(def cluster-name->cluster-dict-atom (atom nil))
+
+(defn cluster-name->db-id
+  "Given a cluster name, return the db-id we should use to refer to that compute cluster
+  when we put it within a task structure."
+  [cluster-name]
+  (let [{:keys [db-id]} (get @cluster-name->cluster-dict-atom cluster-name)]
+    ; All clusters referenced by name must have been installed in the db previously.
+    (when-not db-id (throw (IllegalStateException. (str "Was asked to lookup db-id for " cluster-name " and got nil"))))
+    db-id))
+
+(defn setup-cluster-map-config
+  "Setup the cluster-map configs, linking a cluster name to the associated metadata needed
+  to represent/process it."
+  [conn settings]
+  (let [compute-clusters (->> (get-mesos-clusters-from-config settings)
+                              (map (partial get-mesos-cluster-map conn)))
+        reduce-fn (fn [accum {:keys [compute-cluster-name] :as cluster-dict}]
+                    (when (contains? accum compute-cluster-name)
+                      (throw (IllegalArgumentException. (str "Multiple compute-clusters have the same name: " compute-cluster-name))))
+                    (assoc accum compute-cluster-name cluster-dict))]
+    (map #(log/info "Setting up compute cluster: " %) compute-clusters)
```

Comment:
Does this actually work? I have been burned by lazy `map` before when logging.

Reply:
It actually worked, but doseq is safer. Switched.